api/server/server.go

```diff
@@ -899,14 +899,27 @@ func (s *Server) getImagesGet(eng *engine.Engine, version version.Version, w htt
 	if version.GreaterThan("1.0") {
 		w.Header().Set("Content-Type", "application/x-tar")
 	}
-	var job *engine.Job
+	output := utils.NewWriteFlusher(w)
+
+	imageExportConfig := &graph.ImageExportConfig{
+		Engine:    eng,
+		Outstream: output,
+	}
 	if name, ok := vars["name"]; ok {
-		job = eng.Job("image_export", name)
+		imageExportConfig.Name = append(imageExportConfig.Name, name)
 	} else {
-		job = eng.Job("image_export", r.Form["names"]...)
+		imageExportConfig.Name = r.Form["names"]
 	}
-	job.Stdout.Add(w)
-	return job.Run()
+
+	if err := s.daemon.Repositories().ImageExport(imageExportConfig); err != nil {
+		if !output.Flushed() {
```

Comment:
Is this trick really necessary here? I think it will be enough just return error. At least it'll preserve old behavior.